packages/web-app-files/src/services/folder/loaderSpace.ts

```diff
@@ -102,4 +132,77 @@ export class FolderLoaderSpace implements FolderLoader {
       }
     }).restartable()
   }
+
+  /**
+   * Gets the drive item for a given shared space.
+   */
+  private async getSharedDriveItem({
+    graphClient,
+    spacesStore,
+    space
+  }: {
+    graphClient: Graph
+    spacesStore: SpacesStore
+    space: SpaceResource
+  }) {
+    await spacesStore.loadMountPoints({ graphClient })
+
+    const mountPoints = spacesStore.spaces.filter(isMountPointSpaceResource)
+    // even if the resource has been shared via multiple permissions (e.g. directly via user and a group)
+    // we only care about one matching mount point since the remote item contains all permissions
+    const matchingMountPoint = mountPoints.find((s) => s.root?.remoteItem?.id === space.id)
+    if (!matchingMountPoint) {
+      return null
+    }
+    const { id } = matchingMountPoint
+    return graphClient.driveItems.getDriveItem(id.split('!')[0], id)
+  }
+
+  /**
+   * Since shared spaces are only virtual, they and their permissions can't be fetched from the server.
+   * Hence the permissions for the current user need to be set manually via the corresponding drive item.
+   */
+  private setCurrentUserShareSpacePermissions({
+    sharesStore,
+    spacesStore,
+    userStore,
+    space,
+    sharedDriveItem
+  }: {
+    sharesStore: SharesStore
+    spacesStore: SpacesStore
+    userStore: UserStore
+    space: SpaceResource
+    sharedDriveItem: DriveItem
+  }) {
+    const permissions = sharedDriveItem?.remoteItem?.permissions || []
+    if (!permissions.length) {
+      return
+    }
+
+    const allPermissions: string[] = []
+    permissions.forEach((permission) => {
+      if (permission['@libre.graph.permissions.actions']) {
+        allPermissions.push(...permission['@libre.graph.permissions.actions'])
+        return
+      }
+      const role = sharesStore.graphRoles[permission.roles[0]]
+      if (role) {
```

Comment:
rather have an early return here, which translates to 'continue' in a regular for each

Reply:
Why? 😅 The if-block has 2 simple lines of code, so it wouldn't enhance readability or save up anything.

Reply:
For me it indeed improves readability 🤭